stackrox/stackrox
# ROX-22238: Introduce roxctl deployment check remarks

## Description

This PR adds new output  between roxctl and Central.
Centrals answer to `roxctl deployment check` can now contain `Remarks`, similar to Scanners notes.
These remarks contain for each checked deployment: 
- The permission level the deployment would run as
- Network Policies that would be applied, would the deployment run in the specified cluster/namespace.
The additional information will always be displayed in JSON output. In the table output, the user has to provide `--verbose` explicitly, to keep the default output clean and clutter-free.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed
- Deployed PR locally
- Set up API token for roxctl
- Call `roxctl deployment check --file ~/Downloads/nginx.yaml --cluster remote --namespace stackrox`
  - Observe *no change to the human readable output*
- Call `roxctl deployment check --file ~/Downloads/nginx.yaml --cluster remote --namespace stackrox --verbose`
  - Observe remarks being printed out for checked deployments

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

---
## Additional material
The JSON output can now be found in the `common/report` test cases and newly added JSON files, e.g. [this one](https://github.com/stackrox/stackrox/pull/9749/files#diff-55e4ff8d13824ce8efaef2ccd8ce004bc402ef945b467ca62acc3279288a6e03).
Additionally, here is a screenshot of a deployment check with an applied Network Policy. Note the additional remarks section under the table.

![image](https://github.com/stackrox/stackrox/assets/8026915/667722ee-016e-4cd3-a304-70d47a884a84)
